libsplinter/src/threading/pacemaker.rs

```diff
@@ -20,28 +20,31 @@ use std::sync::{
 use std::thread;
 use std::time::Duration;
 
-use crate::network::connection_manager::error::ConnectionManagerError;
+use super::error::PacemakerStartError;
 
+/// Pacemaker is responsible for periodically sending a message to
+/// another component over a channel. The message is meant to be used as
+/// a notfication that some action should take place.
 pub struct Pacemaker {
     interval: u64,
     join_handle: Option<thread::JoinHandle<()>>,
-    shutdown_handle: Option<ShutdownHandle>,
+    shutdown_signaler: Option<ShutdownSignaler>,
 }
 
 impl Pacemaker {
     pub fn new(interval: u64) -> Self {
```

Comment:
If it's not too much work, can you add doc comments for public structs/methods in the module?

Reply:
will be added in a future PR